### PR TITLE
feat(subservices): add retrieve for subservices to delete on migration

### PR DESCRIPTION
**Pull request by Steffy29**
- retrieve subservices to delete with the selected options on the migration offers page

ref: XDSL-5202

### CODE REVIEW

**Comment by jleveugle** [comment]
Hello @Steffy29,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Comment by Steffy29** [comment]
This PR is linked with https://github.com/ovh-ux/ovh-api-services/pull/118

File: client/telecomUniverseComponents/telecom/pack/migration/pack-migration-process.service.js [hunk 1]
```diff
@@ -232,6 +232,32 @@ export default /* @ngInject */ function ($q, OvhApiPackXdsl, Poller) {
...
+    =           SERVICES TO DELETE        =
+    ======================================= */
+  self.getServicesToDelete = function () {
+    // options post params
+    const migrationOptions = self.getOptionsSelected().map(option => ({
+      name: option.name,
+      quantity: option.included,
+    }));
+
+    if (migrationProcess.selectedOffer.options.gtr_ovh
```
Comment (FredericEspiau) [inline client/telecomUniverseComponents/telecom/pack/migration/pack-migration-process.service.js]:
```suggestion
    if (_.get(migrationProcess, 'selectedOffer.options.gtr_ovh.selected'))
```

File: client/app/telecom/pack/migration/service-delete/telecom-pack-migration-service-delete.controller.js [hunk 2]
```diff
@@ -30,8 +30,22 @@ angular.module('managerApp').controller('TelecomPackMigrationServiceDeleteCtrl',
...
+        _.set(mysubService, 'services', subService.services.map((service, index, originalArray) => ({
+          name: service,
+          selected: originalArray.length === subService.numberToDelete,
+        })));
+        subServicesToDelete.push(mysubService);
+      });
+      self.process.selectedOffer.subServicesToDelete = subServicesToDelete;
+
+      self.chunkedSubServices = _.chunk(self.process.selectedOffer.subServicesToDelete, 2);
+    }, (error) => {
```
Comment (FredericEspiau) [inline client/app/telecom/pack/migration/service-delete/telecom-pack-migration-service-delete.controller.js]:
```suggestion
    })
    .catch((error) => {
```

File: client/app/telecom/pack/migration/service-delete/telecom-pack-migration-service-delete.controller.js [hunk 3]
```diff
@@ -30,8 +30,22 @@ angular.module('managerApp').controller('TelecomPackMigrationServiceDeleteCtrl',
 
   function init() {
     self.process = TucPackMigrationProcess.getMigrationProcess();
-
-    self.chunkedSubServices = _.chunk(self.process.selectedOffer.subServicesToDelete, 2);
+    TucPackMigrationProcess.getServicesToDelete().then((servicesToDelete) => {
+      const subServicesToDelete = [];
+      angular.forEach(servicesToDelete, (subService) => {
```
Comment (FredericEspiau) [inline client/app/telecom/pack/migration/service-delete/telecom-pack-migration-service-delete.controller.js]: `forEach` + `push` === `map` :)

```suggestion
     const subServicesToDelete = servicesToDelete.map((subService) => {
```

**Comment by Steffy29** [comment]
For the moment, this feature is not needed.

**Pull request closed without merging**